libflux/flux-core/src/semantic/formatter/tests.rs

```diff
@@ -0,0 +1,118 @@
+use crate::semantic::convert_source;
+use crate::semantic::formatter::format;
+use pretty_assertions::assert_eq;
+
+#[test]
+fn literals() {
+    let sem_pkg = convert_source(
+        r#"
+            a = "Hello, World!"
+            b = 12
+            c = 18.5
+            d = 12h
+            e = 2019-10-31T00:00:00Z
+            f = /server[01]/
+        "#,
+    )
+    .unwrap();
+
+    let got = format(&sem_pkg).unwrap();
+
+    let want = r#"package main
+a = "Hello, World!":string
+b = 12:int
+c = 18.5:float
+d = 043200000000000false:duration
+e = 2019-10-31T00:00:00Z:datetime
+f = /server[01]/:regex"#;
+
+    assert_eq!(want, got);
```

Comment:
I would highly recommend using https://docs.rs/expect-test/1.1.0/expect_test/ for tests like these. Makes it much easier to update the tests if formatting changes in the future and skips the need to write the expected value in the first place.